05.asynchronous_sqlite3/callback.js

```diff
@@ -0,0 +1,27 @@
+import sqlite3 from "sqlite3";
+const db = new sqlite3.Database(":memory:");
+
+db.run("CREATE TABLE books (title TEXT)", () => {
+  const stmt = db.prepare("INSERT INTO books VALUES (?)");
+  for (let i = 0; i < 10; i++) {
+    stmt.run("title" + (i + 1));
+  }
```

Comment:
`run()` は非同期に実行されるので、各ループの `run()` が直列に実行されない可能性がある点は認識していますか？ 直列に実行したいのなら、前の `run()` の完了を待つ必要があります。以下のようなコードで再現できます。

```suggestion
  for (let i = 0; i < 10; i++) {
    setTimeout(() => {
      stmt.run("title" + (i + 1));
    }, Math.random * 10);
  }
```

Reply:
認識していませんでした。
ありがとうございます。db.run(INSERT文)をコールバックでつなぐ形に修正しました。